lucene/core/src/java/org/apache/lucene/util/crypto/EncryptedFileChannel.java

```
@@ -0,0 +1,312 @@
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public int write(ByteBuffer src) throws IOException {
+    if (this.cipher == null) {
+      // in case of empty readonly channel - let it crash on write
+      return this.channel.write(src);
+    }
+    ensureIv();
+    long position = this.position();
+    assert position >= 0 : "Cannot write EncryptedFileChannel before IV: " + this.channel.position();
+    byte[] tmp = this.cipher.encrypt(src, position);
+    
+    return this.channel.write(ByteBuffer.wrap(tmp));
+  }
+
+  @Override
+  public int write(ByteBuffer src, long position) throws IOException {
+    if (this.cipher == null) {
+      // in case of empty readonly channel - let it crash on write
+      return this.channel.write(src, position);
+    }
+    ensureIv();
+    byte[] tmp = this.cipher.encrypt(src, position);
+    
+    return this.channel.write(ByteBuffer.wrap(tmp), position + ivLength);
+  }
+
+  @Override
+  public long write(ByteBuffer[] srcs, int offset, int length) throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public long position() throws IOException {
+    return this.channel.position() - ivLength;
+  }
+
+  @Override
+  public FileChannel position(long newPosition) throws IOException {
+    this.channel.position(newPosition + ivLength);      
+    return this;
+  }
+
+  @Override
+  public long size() throws IOException {
+    long size = this.channel.size() - ivLength;
+    assert size >= 0: "Invalid channel state, size is " + size;
+    return size;
+  }
+
+  @Override
+  public FileChannel truncate(long size) throws IOException {
+    this.channel.truncate(size + ivLength);      
+    return this;
+  }
+
+  @Override
+  public void force(boolean metaData) throws IOException {
+    this.channel.force(metaData);
+  }
+
+  @Override
+  public long transferTo(long position, long count, WritableByteChannel target) throws IOException {
```

I'm not entirely sure these methods (`transferTo` and `transferFrom`) are actually used by ElasticSearch. Maybe we can try to throw `UnsupportedOperationException` and see if it still works so we don't have to maintain unnecessary code. Not necessarily as part of this PR but in the future.

---

I've seen this used during debugging (either lucene or elastic)